pkg/http/handler/timeout.go

```diff
@@ -75,25 +78,90 @@ func (h *timeToFirstByteTimeoutHandler) ServeHTTP(w http.ResponseWriter, r *http
 		h.handler.ServeHTTP(tw, r.WithContext(ctx))
 	}()
 
-	timeout := getTimer(h.timeout)
-	var timeoutDrained bool
-	defer func() { putTimer(timeout, timeoutDrained) }()
+	firstByteTimeout := getTimer(h.firstByteTimeout)
+	var firstByteTimeoutDrained bool
+
+	var idleTimeout *optionalTimer
+	if h.idleTimeout < 0 {
+		idleTimeout = newOptionalTimer(nil)
```

Comment:
I see what you did here, and I think this works. However, please allow me to propose an alternative that I personally find more straightforward to read and follow through. Feel free to ignore it

https://github.com/knative/serving/commit/d3f245ed825418fbedcfac7bb52b6e97680d4475

Reply:
Thanks, I like the simplicity of it and I applied it. Now the PR is good for review. Should I squash all the commits? I noticed that the automation seemed to do that when merging.